src/server/routes/monthData.js

```diff
@@ -28,7 +33,35 @@ function selectMonthData(req, res, next) {
   next();
 }
 
+/**
+ * selectClientMonthData(req, res, next)
+ * Gets a specific month's or season's staff pick list on the client side.
+ */
+function selectClientMonthData(req, res) {
+  // only 2017-01 works currently. Comment out the dynamice API link below
+  // nyplApiClientGet(`/book-lists/staff-picks/${req.params.month}`)
+  nyplApiClientGet('/book-lists/staff-picks/2017-01')
+    .then(data => {
+      res.json({
+        title: data.title,
+        date: data.date,
+        currentPicks: {
+          picks: data.picks,
+        },
+      });
+    })
+    .catch(error => {
+      console.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}`);
+
+      res.json({
+        statusCode: error.statusCode,
+        errorMessage: error.code,
+      });
+    });
+}
+
 export default {
   currentMonthData,
   selectMonthData,
+  selectClientMonthData,
```

Comment:
Just curious here, but why the name change to selectClientMonthData? Also, don't we need to make a route change for lists so that /staff-picks/2017-01 still provides the requested list? Individual book pages are going to need /staff-picks/2017-01/[isbn-title] routes in order to provide backwards compatibility, so perhaps the list routes also need to exist for the same.

When the selector is changed, the list updates just fine which is great, but the route does not include the URL param for its identifier `2017-01`. Am I incorrect in my thinking?

Reply:
The name here is because it  updates month data but not like `selectMonthData`, it handles the requests from client side. I am actually open to a better name. Just some apps we named these two kind of functions that way.

And the second question is yes we should change the URL. I will do it the next ticket with handling server side API call.